src/Fusion.Summary.Functions/ReportCreator/WeeklyTaskOwnerReportDataCreator.cs

```diff
@@ -0,0 +1,175 @@
+    private static bool IsSupportPosition(ApiPositionV2 position)
+    {
+        var supportNames = new[] { "support", "advisor", "assistance" };
+        return supportNames.Any(s => position.Name.Contains(s, StringComparison.OrdinalIgnoreCase));
+    }
+
+    public static List<ExpiringPosition> GetPositionAllocationsEndingNextThreeMonths(IEnumerable<ApiPositionV2> allProjectPositions)
+    {
+        /*
+         * Remember that it's the position*Allocation* that is expiring, not the position itself.
+         * So a position allocation can be considered expiring if:
+         * 1. The position is active and the next split within the next 3 months does not have a person assigned
+         *      - We want to notify task owners that an upcoming split is missing an allocation and that they should assign someone
+         *
+         * 2. The last split is expiring within the next 3 months.
+         *    There can be more splits after this but if they're not starting (appliesFrom) within the next 3 months (from NowDate), we consider the position expiring.
+         *    Once the later split comes within the 3-month window, the position will fall under TBNPositionsStartingWithinThreeMonths
+         *      - We want to notify task owners that the position is expiring
+         *
+         * Note: If there is a gap between two splits and the gap/time-period is fully within the 3-month window,
+         * then we DO NOT consider the position expiring. This is also an unusual case.
+         */
+
+        var nowDate = NowDate;
+        var expiringDate = nowDate.AddMonths(3);
+
+        var expiringPositions = new List<ExpiringPosition>();
+
+        foreach (var position in allProjectPositions)
+        {
+            if (position.Instances.Count == 0) // No instances, skip
+                continue;
+
+            var activeInstance = position.Instances.FirstOrDefault(i => i.AppliesFrom <= nowDate && i.AppliesTo >= nowDate);
+
+
+            // Find future instances with a start date within the 3-month window that may or may not end within the 3-month window
+            var futureInstances = position.Instances
+                .Where(i => i.AppliesFrom >= nowDate && i.AppliesFrom < expiringDate)
+                .OrderBy(i => i.AppliesFrom)
+                .ToList();
+
+            // Handle case where the position is not currently active
+            if (activeInstance is null)
+            {
+                if (futureInstances.Count == 0)
+                    continue; // This is a past position
+
+                var endingPositionAllocation = FindFirstTBNOrLastExpiringInstance(futureInstances);
+
+                // If the last instance is not the last instance then there are more instances after it that are not within the 3-month window or TBN
+                var isEndingInstanceLast = futureInstances.Last() == endingPositionAllocation;
```

Comment:
A bit nitpick, but the comment related to this is some what unclear.
But it also makes me think about the cases where we have several instances that starts after today and ends within 3 months where some are allocated, and some are not. This should be an edge case of course.

Reply:
Updated comment and added another test